src/lib/snack-bar/snack-bar.ts

```diff
@@ -145,6 +148,18 @@ export class MatSnackBar {
     // We can't pass this via the injector, because the injector is created earlier.
     snackBarRef.instance = contentRef.instance;
 
+    // Subscribe to the breakpoint observer and attacg the mat-snack-bar-handset class as
+    // appropriate. This class is applied to the overlay element because the overlay must expand to
+    // fill the width of the screen for full width snackbars.
+    const isHandset = this._breakpointObserver.observe(Breakpoints.Handset).subscribe(state => {
+      overlayRef.overlayElement.classList.toggle('mat-snack-bar-handset', state.matches);
+    });
+    // Unsubscribe from the isHandset subscription when the overlay is detached from the view.
+    const overlayDetach = overlayRef.detachments().subscribe(() => {
+      overlayDetach.unsubscribe();
```

Comment:
A cleaner approach could be to use `takeUntil`. E.g.

```
this._breakpointObserver.observe(Breakpoints.Handset)
  .takeUntil(overlayRef.detachments().first())
  .subscribe(() => ...);
```

This way you don't have to keep track of subscriptions.

Reply:
Good call